Doc/howto/clinic.rst

```diff
@@ -93,67 +96,148 @@ and it should be able to do many interesting and smart
-line.  However, if the input hasn't changed, the output won't change either.
-
-You should never modify the output portion of an Argument Clinic block.  Instead,
-change the input until it produces the output you want.  (That's the purpose of the
-checksum—to detect if someone changed the output, as these edits would be lost
-the next time Argument Clinic writes out fresh output.)
-
-For the sake of clarity, here's the terminology we'll use with Argument Clinic:
-
-* The first line of the comment (``/*[clinic input]``) is the *start line*.
-* The last line of the initial comment (``[clinic start generated code]*/``) is the *end line*.
-* The last line (``/*[clinic end generated code: checksum=...]*/``) is the *checksum line*.
-* In between the start line and the end line is the *input*.
-* In between the end line and the checksum line is the *output*.
-* All the text collectively, from the start line to the checksum line inclusively,
-  is the *block*.  (A block that hasn't been successfully processed by Argument
-  Clinic yet doesn't have output or a checksum line, but it's still considered
-  a block.)
+will discard the old :term:`output` and write out the new output with a fresh
+:term:`checksum line`.
+If the :term:`input` hasn't changed, the output won't change either.
+
+.. note::
+
+   You should never modify the output of an Argument Clinic block,
+   as any change will be lost in future Argument Clinic runs;
+   Argument Clinic will detect an output checksum mismatch and regenerate the
+   correct output.
+   If you are not happy with the generated output,
+   you should instead change the input until it produces the output you want.
+
+
+.. _clinic-reference:
+
+Reference
+=========
+
+
+.. _clinic-terminology:
+
+Terminology
+-----------
+
+.. glossary::
+
+  start line
+     The line ``/*[clinic input]``.
+     This line marks the beginning of Argument Clinic input.
+     Notice that the *start line* opens a C block comment.
+
+  end line
+     The line ``[clinic start generated code]*/``.
+     This line marks the end of Argument Clinic input.
+     Notice that the *end line* closes the C block comment opened
+     by the *start line*.
+
+  checksum line
+     A line that looks like ``/*[clinic end generated code: ...]*/``.
+     The three dots will be replaced by the actual checksum(s).
+     The checksum line marks the end of Argument Clinic generated code.
+
+  input
+     The text between the :term:`start line` and the :term:`end line`.
+     Notice that the start and end lines open and close a C block comment;
+     the *input* is thus a part of that same C block comment.
+
+  output
+     The text between the :term:`end line` and the :term:`checksum line`.
+
+  block
+     All text from the :term:`start line` to the :term:`checksum line` inclusively.
+
+.. note::
+
+   A block that hasn't been successfully processed by Argument Clinic yet
+   doesn't have *output* or a *checksum line*,
+   but it's still considered a *block*.
```

Comment:
Surely this contradicts the definition of "block" that's given immediately above?

Reply:
I'm not sure how to address this, yet.

Reply:
Contributing to the confusion, *input* is sometimes referred to as the *input block*; ditto for *output*.

Reply:
I'm not sure what this note actually adds for the user. It seems to be an implementation detail exposed in the docs.

Reply:
I'll remove this note before landing. (On mobile now.)

Reply:
Agree that the note doesn't add too much, but I think it's much better now either way!